blockly-core/core/ui/fields/field_parameter.js

```diff
@@ -68,11 +68,13 @@ Blockly.FieldParameter.dropdownChange = function(text) {
   var oldVar = this.getText();
   if (text === Blockly.Msg.RENAME_PARAMETER) {
     this.getParentEditor_().hideChaff();
-    text = Blockly.FieldVariable.promptName(Blockly.Msg.RENAME_PARAMETER_TITLE.replace('%1', oldVar),
-      oldVar);
-    if (text) {
-      Blockly.Variables.renameVariable(oldVar, text, this.sourceBlock_.blockSpace);
-    }
+    Blockly.FieldVariable.modalPromptName(
+        Blockly.Msg.RENAME_PARAMETER_TITLE.replace('%1', oldVar),
+        Blockly.Msg.CONFIRM_RENAME_VARIABLE,
+        oldVar,
+        function(text) {
```

Comment:
nit: might be good to rename this somethings like `newVar`, just because `text` is already a parameter of this function